### ISSUE TITLE
Ordinal scale turns into point scale when used for opacity encoding

### ISSUE THREAD

**Issue by pgorczak** [vega/vega-lite#5982]
It seems like ordinal scales used for opacity encoding get overwritten into point scales when compiling to Vega. The [docs](https://vega.github.io/vega-lite/docs/scale.html#discrete) mention that point scales are the default for opacity encoding on discrete fields, but I think it would be nice if using ordinal scales was also possible. It does work when manually tweaking the Vega output (see below).

Vega-Lite specification thanks to @jakevdp  ([vega editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4zEgVwE4BsVTkQHoBjWLAZwDoArCgewDtIBfAGnCgFtyBrfSFgCmJeJA4RIQxiXrQAlowDm+UBEn0ADrBLz4AT1Wd1UA5qEDG9Loth4JJqADN5QnNAEB5LPKWLxxuqQFGQ4FqhqjpLQ1rD+qADakACi2FoWbFAAUrDazJmQAKoAygCCkAC6DlGCsMrhYAkAjJkADFQtYE1VgSaQZg2Q9FgKjHaQvWAsgeyBkIYR-frmAsOj4wUubh6okN6+-rN9AB6qkLBKSsJKCIOyGIxiBQMCAI4YdfB6CPIAbhbTCDTFhAA))

```json
{
  "data": {"url": "data/cars.json"},
  "mark": "rect",
  "encoding": {
    "opacity": {
      "field": "Origin",
      "type": "nominal",
      "scale": {
        "domain": ["Europe", "Japan", "USA"],
        "range": [1, 0.1, 1],
        "type": "ordinal"
      }
    },
    "y": {"field": "Origin", "type": "ordinal"},
    "x": {"aggregate": "count", "type": "quantitative"}
  }
}
```

### Resulting output and Vega scale spec

![](https://user-images.githubusercontent.com/1765698/75535132-6398e900-5a15-11ea-93f5-c38a847636a1.png)

```json
{
  "name": "opacity",
  "type": "point",
  "domain": ["Europe", "Japan", "USA"],
  "range": [1, 0.1, 1]
}
```

### Expected output and (manually fixed) Vega scale spec

![](https://user-images.githubusercontent.com/1765698/75536001-a3f86700-5a15-11ea-95ec-7b47e572a50b.png)

```json
{
  "name": "opacity",
  "type": "ordinal",
  "domain": ["Europe", "Japan", "USA"],
  "range": [1, 0.1, 1]
}
```

**Comment by domoritz** [comment]
Just noting that this visual encoding has some perception issues which is why we so far made the choice to not allow nominal opacity encodings.

```
[Warning] Channel opacity should not be used with an unsorted discrete field.
[Warning] Channel "opacity" does not work with "ordinal" scale. We are using "point" scale instead.
```

Having said that, the manual scale should take precedence here if it works.

**Comment by pgorczak** [comment]
FWIW I ran into this while trying to layer low opacity area/bar marks with high opacity point/line marks ([example][1]) with a nice legend. I just played around a bit more and noticed that opacity is not reflected in the legend when encoding both color and opacity ([example][3]). I guess it's not really related to this so I'll open a separate issue.

[1]: https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4zEgVwE4BsVTkQHpYA7AZwGMB7AWwCMBTAOgCtzrTIBfAGnCiNSNaAEtSAc3ygIESAA9pkeAE8ADo3yQAjhjLxRiAwDdNvKADNRjHNC0ANHv1lQVS1Rq279hhKNOQ5pBWNnaokACaTgJy1GqwlIZuqDIuUB6a4aR04rB4zmnB1rZaAMqMWNbkgTGykFR5mWi1LjB0sOL4ANqQAJKBUL39ALoFaVBYZBJNXQCM5gAMzADMoy11GVrUWGKkeZDr3LVHEHwCkDiwKhXdtalytLBYANZaT4xIY+mTFBbbtN10FYcPAbuFCBhaMxypVGOQwABeBFgADkQxRPGGxy+9ygjxeWjU1HE8BqrXgP3IfywANQXSBohBYIICEh0IqVURyLRGO4WJcJzAWO4QA
[3]: https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAmCGAutIC4zEgVwE4BsVTkQHpYA7AZwGMB7AWwCMBTAOgCtzrTIBfAGnCiNSNaAEtSAc3ygIESAA9pkeAE8ADo3yQAjhjLxRiAwDdNvKADNRjHNC0ANHv1lQVS1Rq279hhKNOQ5pBWNnaokACaTgJyNDjUWNIxssrqmuGkdOKweM4ulta2WgDKjFjW5IHJclQ56WgwdLDi+ADakACSgVAdXQC63Ml8yZDUarCUhm6oMvmpnhlZpDlV+QWhJWUVq3O1OPWza420zVyo7V1BvZB9eUdYZBL1rQCM5gAMzADMt9UpHvVRlgxMs8H9Bi4IWBhnIcLAVGU2slDlATlgANZaWBYRhIO5QeAPCgWBK0NroKw4eCI8KEDC0ZilcqMchgAC8bLAAHJelyeH0hviUZA0ZjwmpqOJ4Ds5ISyOQSVgyecKaIqTSCAh6Yytiz2ZyeXzuALIQIBdwgA
